add-on/src/lib/redirect-handler/blockOrObserve.ts

```diff
@@ -281,8 +282,10 @@ function saveAndGenerateRule (
   regexSubstitution: string,
   excludedInitiatorDomains: string[] = []
 ): browser.DeclarativeNetRequest.Rule {
-  // We need to generate a random ID for the rule.
-  const id = Math.floor(Math.random() * 29999)
+  // We need to generate a positive number as an id.
+  const id = fastHashCode(`${regexFilter}:${regexSubstitution}:${excludedInitiatorDomains.join(':')}`, {
```

Comment:
This seems reasonable, but the link provided by `fast-hash-code` to the implementation reasoning is no longer available. It makes me wonder about collisions. `webCrypto.subtle.digest` should do what we want and is available in service workers, is there a reason we didn't use that?

does `id` need to be a number?